### ISSUE TITLE
Detach memory inspection panes from main Insight window

### ISSUE THREAD

**Issue by navnavnav** [navnavnav/Bloom#49]
It would be nice if we could detach a particular memory inspection pane from the Insight window, so that it becomes a new window and is no longer constrained to the size of the Insight window.

**Comment by navnavnav** [comment]
Quick update on this:

Development work for this is almost complete.

Memory inspection panes can now be detached from the main window:

![Bloom Insight_135](https://user-images.githubusercontent.com/3252338/179367240-7029b6ce-b114-44f7-8a01-39d47d09c2f9.png)

![Selection_136](https://user-images.githubusercontent.com/3252338/179367280-e7943fca-b0ca-40d2-aeca-b36888ac590b.png)

![Memory Inspection - Internal RAM_137](https://user-images.githubusercontent.com/3252338/179367316-af4ca0a6-d075-4eb0-8c2a-9bb91898d110.png)

To do:

- Persist attached/detached state, as well as detached window size and position, across debug sessions (these properties will be saved in the current project's settings)
- Prevent inspection panes from needlessly refreshing memory data post activation

**Comment by navnavnav** [comment]
Development work for this is complete.

It didn't make sense to have the refresh button in the `HexViewerWidget`, and given that I intend to reuse the `HexViewerWidget` for [snapshot diffs](https://github.com/navnavnav/Bloom/issues/46), I decided to move the button to the `MemoryInspectionPane` widget, which is where it should have been in the first place.

![Memory Inspection - Internal RAM_138](https://user-images.githubusercontent.com/3252338/180309140-de0c4e4b-e5aa-4825-b338-24f2027a79b2.png)

Additional refresh configuration, accessible via a context menu of the refresh tool button (by right-clicking the refresh button):

![Selection_140](https://user-images.githubusercontent.com/3252338/180309452-01add422-1823-4951-a401-f6753e274e86.png)

Also updated the 'Manage Memory Regions' tool button icon, to make it more distinct.

These changes have been merged into develop. They will be part of v0.11.0. Closing this now.
